src/rfc3161_client/verify.py

```diff
@@ -14,15 +14,15 @@
 
 @dataclass
 class VerifyOpts:
-    policy_id: cryptography.x509.ObjectIdentifier | None
-    tsa_certificate: cryptography.x509.Certificate | None
-    intermediates: list[cryptography.x509.Certificate]
-    roots: list[cryptography.x509.Certificate]
-    nonce: int
-    common_name: str
+    policy_id: cryptography.x509.ObjectIdentifier | None = None
+    tsa_certificate: cryptography.x509.Certificate | None = None
+    intermediates: list[cryptography.x509.Certificate] = field(default_factory=list)
+    roots: list[cryptography.x509.Certificate] = field(default_factory=list)
+    nonce: int | None = None
+    common_name: str | None = None
```

Comment:
Thinking about the API design a bit, if all fields are optional, I wonder if a builder API would make more sense? Similar to what `cryptography` does for [x509 verification](https://cryptography.io/en/latest/x509/verification/#cryptography.x509.verification.PolicyBuilder).

Something that ends up looking like:
```python
verifier = VerifierBuilder().nonce(n).common_name(x).build_verifier()
verifier.verify(timestamp_response)
```

cc @woodruffw for a 2nd opinion